cdk_framework/EKS/test/cluster-deployment.test.ts

```diff
@@ -0,0 +1,77 @@
+import * as cdk from 'aws-cdk-lib';
+import { Template } from 'aws-cdk-lib/assertions';
+import { readFileSync, writeFileSync } from 'fs';
+import { validateClustersConfig } from '../lib/utils/validate';
+import { Stack, StackProps, aws_eks as eks, aws_ec2 as ec2} from 'aws-cdk-lib';
+import { ClusterStack } from '../lib/stacks/cluster-stack';
+import { VPCStack } from '../lib/utils/vpc-stack';
+const yaml = require('js-yaml')
+
+// example test. To run these tests, uncomment this file along with the
+// example resource in lib/my-project-stack.ts
+test('ClusterTest', () => {
+    var app = new cdk.App();
+    // WHEN
+    const route = process.env.CDK_CONFIG_PATH ||  __dirname + '/config/clusters.yml';
+    const raw = readFileSync(route)
+    const data = yaml.load(raw)
+
+    var clusterMap = new Map<string, ClusterStack>()
+    var versionMap = new Map<string, string>()
+
+    const vs = new VPCStack(app, "EKSVpc", {
+        env: {
+          region: 'us-west-2'
+        }
+      })
+
+    validateClustersConfig(data)
+    // const bigMap = parseData(data['clusters'])
+    for(const [key, value] of Object.entries(data['clusters'])){
+        const val = Object(value)
+        const versionKubernetes = eks.KubernetesVersion.of(String(val['version']));
+        const newStack = new ClusterStack(app, key + "Stack", {
+            launchType: String(val['launch_type']),
+            name: key,
+            vpc: vs.vpc,
+            version: versionKubernetes,
+            cpu: String(val["cpu_architecture"]),
+            nodeSize: String(val["node_size"]),
+            env: {
+            region: 'us-west-2'
+            },
+        })
+
+        clusterMap.set(key, newStack)
+        versionMap.set(key, String(val['version']))
+
+    
+    }
+
+    for(const [key, st] of clusterMap){
+        var template = Template.fromStack(st);
+        var KubernetesVersion = versionMap.get(key)
+
+        template.hasResourceProperties("Custom::AWSCDK-EKS-Cluster", {
+            Config: {
+                name: key,
+                version: KubernetesVersion,
+                logging: {
+                  clusterLogging: [
+                    {
+                      enabled: true,
+                      types: [
+                        "api",
+                        "audit",
+                        "authenticator",
+                        "controllerManager",
+                        "scheduler"
+                      ]
+                    }
+                  ]
+                }
+            }
+        })
```

Comment:
Can we test that node groups were successfully added with the correct size?

Reply:
The closest result is:
      {
        "Type": "Custom::AWSCDK-EKS-Cluster",
        "Properties": {
          "ServiceToken": {
            "Fn::GetAtt": [
              "awscdkawseksClusterResourceProviderNestedStackawscdkawseksClusterResourceProviderNestedStackResource9827C454",
              "Outputs.amdCluster12StackawscdkawseksClusterResourceProviderframeworkonEventB79DA207Arn"
            ]
          },
          "Config": {
            "name": "amdCluster12",
            "version": "1.21",
            "roleArn": {
              "Fn::GetAtt": [
                "amdCluster12RoleA3841D28",
                "Arn"
              ]
            },
            "resourcesVpcConfig": {
              "subnetIds": [
                {
                  "Fn::ImportValue": "EKSVpc:ExportsOutputRefEKSVpcpublicsubnetSubnet1Subnet66FF0BF6D340A731"
                },
                {
                  "Fn::ImportValue": "EKSVpc:ExportsOutputRefEKSVpcpublicsubnetSubnet2SubnetCB24423F14E7B916"
                },
                {
                  "Fn::ImportValue": "EKSVpc:ExportsOutputRefEKSVpcpublicsubnetSubnet3Subnet67FAEF799591D405"
                }
              ],
              "securityGroupIds": [
                {
                  "Fn::GetAtt": [
                    "amdCluster12ControlPlaneSecurityGroup84422F7C",
                    "GroupId"
                  ]
                }
              ],
              "endpointPublicAccess": true,
              "endpointPrivateAccess": true
            },
            "logging": {
              "clusterLogging": [
                {
                  "enabled": true,
                  "types": [
                    "api",
                    "audit",
                    "authenticator",
                    "controllerManager",
                    "scheduler"
                  ]
                }
              ]
            }
          },
          "AssumeRoleArn": {
            "Fn::GetAtt": [
              "amdCluster12CreationRoleC962D238",
              "Arn"
            ]
          },
          "AttributesRevision": 2
        },
        "DependsOn": [
          "amdCluster12CreationRoleDefaultPolicy5A28CD29",
          "amdCluster12CreationRoleC962D238"
        ],
        "UpdateReplacePolicy": "Delete",
        "DeletionPolicy": "Delete"
      }

Reply:
This is what the template looks like, so I don't know if it could be attained, but I'll look.